src/components/connection-prompt/ConnectDialogContainer.svelte

```diff
@@ -88,61 +87,79 @@
+    if (!friendlyName) {
+      return;
+    }
+    return flashMicrobit(friendlyName);
+  }
 
-        const hexForStage = stageToHex(flashStage);
-        Microbits.flashHexToLinked(hexForStage, progress => {
-          // Flash hex
-          // Send users to download screen
-          if (
-            $connectionDialogState.connectionState != ConnectDialogStates.USB_DOWNLOADING
-          ) {
-            $connectionDialogState.connectionState = ConnectDialogStates.USB_DOWNLOADING;
-          }
-          flashProgress = progress;
-        })
-          .then(() => {
-            // Finished flashing successfully
-            if (flashStage === 'bluetooth' || flashStage === 'radio-sender') {
-              $connectionDialogState.connectionState =
-                ConnectDialogStates.CONNECT_BATTERY;
-            } else if (flashStage === 'radio-bridge') {
-              onConnectingSerial();
-            }
-          })
-          .catch(err => {
-            if (flashStage === 'bluetooth') {
-              $connectionDialogState.connectionState =
-                ConnectDialogStates.MANUAL_TUTORIAL;
-            } else {
-              handleWebUSBError(err);
-            }
-          });
-      })
-      .catch(err => {
-        if (flashStage === 'bluetooth') {
-          $connectionDialogState.connectionState = ConnectDialogStates.MANUAL_TUTORIAL;
-        } else {
-          handleWebUSBError(err);
+  async function getMicrobitName(): Promise<string | undefined> {
+    try {
+      const friendlyName = await Microbits.getLinkedFriendlyName();
+      // Find the name of the micro:bit
+      if ($connectionDialogState.deviceState === DeviceRequestStates.OUTPUT) {
+        btPatternOutput.set(MBSpecs.Utility.nameToPattern(friendlyName));
+      } else {
+        btPatternInput.set(MBSpecs.Utility.nameToPattern(friendlyName));
+      }
+      return friendlyName;
+    } catch (err) {
+      if (flashStage === 'bluetooth') {
+        $connectionDialogState.connectionState = ConnectDialogStates.MANUAL_TUTORIAL;
+      } else {
+        handleWebUSBError(err);
+      }
+    }
+  }
+
+  async function flashMicrobit(friendlyName: string): Promise<void> {
+    const hexForStage = stageToHex(flashStage);
+    try {
+      await Microbits.flashHexToLinked(hexForStage, progress => {
+        // Flash hex
+        // Send users to download screen
+        if (
+          $connectionDialogState.connectionState != ConnectDialogStates.USB_DOWNLOADING
+        ) {
+          $connectionDialogState.connectionState = ConnectDialogStates.USB_DOWNLOADING;
         }
+        flashProgress = progress;
       });
+      // Finished flashing successfully
+      if (flashStage === 'bluetooth' || flashStage === 'radio-sender') {
+        $connectionDialogState.connectionState = ConnectDialogStates.CONNECT_BATTERY;
+      } else if (flashStage === 'radio-bridge') {
+        onConnectingSerial(friendlyName);
+      }
```

Comment:
I think would be good to somehow move this out as a separate function e.g. `onFlashingFinished` since this is handling after flashing is finished. We can also put the try-catch into the `onConnectingSerial` function since I think that's the only part that might throw. I think the error would be handled by just the `handleWebUSBError` since the `flashstage` is not Bluetooth?